tests/unittests/tests-fib_sr/tests-fib_sr.c

```diff
@@ -0,0 +1,701 @@
+    * expected X0, .., X9
+    */
+    /*
+    fib_print_sr( local_sourceroutes[0]);
+
+    for (size_t i = 0; i < addr_list_elements; ++i) {
+        printf("[");
+        for (size_t j = 0; j < element_size; j++) {
+            printf("%c", addr_list[(i*element_size)+j]);
+        }printf("] ");
+    }puts("");
+    */
+
+    /* remove the source route */
+    TEST_ASSERT_EQUAL_INT(0, fib_sr_delete(local_sourceroutes[0]));
+
+    fib_deinit(NULL);
+}
+
+/*
+* @brief create a new source route with a number of hops on its path,
+*        and get a partial route (also in reverse order)
+*/
+static void test_fib_sr_11_create_sr_with_hops_and_get_a_partial_route(void)
+{
+    fib_sr_t *local_sourceroutes[1];
+    size_t add_buf_size = 16;
+    char addr_nxt[add_buf_size];
+
+    TEST_ASSERT_EQUAL_INT(0, fib_sr_create(&local_sourceroutes[0], 42, 0x0, 10000));
+
+    for (size_t i = 0; i < 10; ++i) {
+        snprintf(addr_nxt, add_buf_size, "Some address X%d", i);
+        /* append a  hop */
+        TEST_ASSERT_EQUAL_INT(0, fib_sr_entry_append(local_sourceroutes[0],
+                              (uint8_t *)&addr_nxt,
+                              add_buf_size)
+                             );
+    }
+
+    size_t addr_list_elements = 10;
+    size_t element_size = 16;
+    uint8_t addr_list[ addr_list_elements * element_size ];
+    kernel_pid_t sr_iface_id;
+    uint32_t sr_flags = 0x0;
+    snprintf(addr_nxt, add_buf_size, "Some address X7");
+
+    TEST_ASSERT_EQUAL_INT(0, fib_sr_get_route((uint8_t *)&addr_nxt, add_buf_size, &sr_iface_id,
+                          &sr_flags, addr_list, &addr_list_elements, &element_size,
+                          false, NULL)
+                         );
+
+    /*
+    * @note uncomment the following lines to print the resulting source route.
+    * expected X0..X9 where X0..X7 has usecount of 2
+    */
+    /*
+    fib_print_sr( local_sourceroutes[0]);
+
+    for (size_t i = 0; i < addr_list_elements; ++i) {
+        printf("[");
+        for (size_t j = 0; j < element_size; j++) {
+            printf("%c", addr_list[(i*element_size)+j]);
+        }printf("] ");
+    }puts("");
+    */
+
+    /* now we get the addresses in reverse order */
+    TEST_ASSERT_EQUAL_INT(0, fib_sr_get_route((uint8_t *)&addr_nxt, add_buf_size, &sr_iface_id,
+                          &sr_flags, addr_list, &addr_list_elements, &element_size,
+                          true, NULL)
+                         );
+
+    /*
+    * @note uncomment the following lines to print the resulting source route.
+    */
+    /*
+    for (size_t i = 0; i < addr_list_elements; ++i) {
+        printf("[");
+        for (size_t j = 0; j < element_size; j++) {
+            printf("%c", addr_list[(i*element_size)+j]);
+        }printf("] ");
+    }puts("");
+    */
+
+    /* remove the source route */
+    TEST_ASSERT_EQUAL_INT(0, fib_sr_delete(local_sourceroutes[0]));
```

Comment:
hmm, isnt `fib_sr_get_route` creating a partial source route when you query it for `"Some address X7"`? So deleting local_sourceroutes[0] will not fully clear the fib?